cli/endpoints/batch/deploy-models/openai-embeddings/code/batch_driver.py

```diff
@@ -0,0 +1,88 @@
+import os
+import glob
+import mlflow
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from typing import List
+from datasets import load_dataset
+
+DATA_READERS = {
+    ".csv": "csv",
+    ".tsv": "tsv",
+    ".parquet": "parquet",
+    ".json": "json",
+    ".jsonl": "json",
+    ".arrow": "arrow",
+    ".txt": "text",
+}
+
+
+def init():
+    global model
+    global output_file
+    global task_name
+    global text_column
+
+    # AZUREML_MODEL_DIR is the path where the model is located.
+    # If the model is MLFlow, you don't need to indicate further.
+    model_path = glob.glob(os.environ["AZUREML_MODEL_DIR"] + "/*/")[0]
+    # AZUREML_BI_TEXT_COLUMN is an environment variable you can use
+    # to indicate over which column you want to run the model on. It can
+    # used only if the model has one single input.
+    text_column = os.environ.get("AZUREML_BI_TEXT_COLUMN", None)
```

Comment:
Why this and no signature? Is it so customers can pass in data that has different column names like "input" vs "input_string", vs "prompt" vs whatever?

Reply:
So signature tells you what’s the thing the model is expecting. All these models expect you to pass input text. However, to avoid customers having to handcraft a dataset to match the name of the columns in the signature, we can just pass them a way to select which is the column they want to send to the model. This will be like the binding. Why? When signatures have only 1 inputs (let’s say text), then the name is not enforced by MLflow. This gives the flexibility of passing any dataframe with a trivial column name and that will work. No need to rename the column to “input text” or “text” or “prompt”.